theano/compile/debugmode.py

```diff
@@ -550,6 +534,7 @@ def get_id_str(obj, get_printed=True):
             id_str = f"[id {char_from_number(len(used_ids))}]"
         elif ids == "":
             id_str = ""
+        assert id_str is not None
```

Comment:
Why was this added?

Reply:
I put it I to make mypy happy, to support my claim that `id_str` was a string, IIRC.

Reply:
Don't worry about `mypy` results; this project doesn't officially support type checking, yet.

Regardless, we need to remove this before we can merge.

Reply:
Hi @brandonwillard -- I'm ok with doing this, but I wonder why you are so firm about it needing to be removed.  Is there some design or style criterion this violates?

Reply:
By the way, in connection with this question, anyone know what the heck the `get_printed` argument is for?  Happy to add doc for this to the PR if someone provides an answer.

Reply:
I'm not familiar enough with this function to know whether or not that `assert` is valid.  For instance, it might be acceptable for `id_str` to be `None`, or perhaps we could be more explicit and avoid an `assert` altogether with an `else` statement that sets `id_str = ""`.  Plus, it's almost always better to raise a more informative `Exception` when/if you expect an expression to be violated.  (Personally, I think the former would be ideal.)

Since this change isn't blocking the PR, it's hard to justify walking through `debugmode.py`'s logic in order to answer these questions.  Also, even if the tests currently pass with this `assert`, I don't believe we're using `debugmode` much at all, so it might not actually be getting the coverage it needs.

Regarding `mypy`, can't you set `id_str: Optional[Text]`?  That would allow `id_str` to be `None`&mdash;according to `mypy`, at least.